internal/controller/model_nodeport.go

```diff
@@ -40,27 +48,51 @@ func NewNodePortLoadBalancerModelGenerator(
 	}
 }
 
-func (g *NodePortLoadBalancerModelGenerator) getDestinationAddresses() (result []string, err error) {
+func isValidIp(ipString string) (err error) {
+	ipParsed := net.ParseIP(ipString)
+	if ipParsed != nil {
+		return nil
+	}
+	return ErrInvalidIpAddress
+}
+
+func isIPv4Address(ipString string) (isIPv4 bool) {
+	return strings.Count(ipString, ":") == 0 && strings.Count(ipString, ".") == 3
+}
+
+func isIPv6Address(ipString string) (isIPv6 bool) {
+	return strings.Count(ipString, ":") >= 2
+}
+
+func (g *NodePortLoadBalancerModelGenerator) getDestinationAddresses() (addresses_v4 []string, addresses_v6 []string, err error) {
```

Comment:
Please don’t use `snake_case` in golang… it is more of a `camelCase` language (re `addresses_v4` etc.)